test/test_theme_assets_reader.rb

```diff
@@ -0,0 +1,61 @@
+require "helper"
+
+class TestThemeAssetsReader < JekyllUnitTest
+  def setup
+    @site = fixture_site(
+      "theme"       => "test-theme",
+      "theme-color" => "black"
+    )
+    assert @site.theme
+  end
+
+  def assert_file_with_relative_path(haystack, relative_path)
+    assert haystack.any? { |f|
+      f.relative_path == relative_path
+    }, "Site should read in the #{relative_path} file, " \
+      "but it was not found in #{haystack.inspect}"
+  end
+
+  def refute_file_with_relative_path(haystack, relative_path)
+    refute haystack.any? { |f|
+      f.relative_path == relative_path
+    }, "Site should not have read in the #{relative_path} file, " \
+      "but it was found in #{haystack.inspect}"
+  end
+
+  context "with a valid theme" do
+    should "read all assets" do
+      ThemeAssetsReader.new(@site).read
+      assert_file_with_relative_path @site.static_files, "assets/img/logo.png"
+      assert_file_with_relative_path @site.pages, "assets/style.scss"
+    end
```

Comment:
If I have `/assets/foo.js` in my site, and `/assets/foo.js` in my theme, to confirm `/assests/foo.js` in the resulting site contain's the site's content, not the theme's.